### ISSUE TITLE
Cannot print Avery 5160 label with correct layout

### ISSUE THREAD

**Issue by ch-gz** [snipe/snipe-it#2555]
I tried several sets of settings. The margin top , left and right looks correct, but horizontal space doesn't change with different value setting. I'm using **chrome** browser. 
Label setting:
Labels per page 30
Label font size 9 pt
Label dimensions (inches)
2.62500 w
1.00000 h

Label spacing (inches)
0.10000 horizontal
0.05000 vertical

Page margins (inches)
0.50000 top
0.18750 right (3/16 in)
0.50000 bottom
0.18750left (3/16 in)

Page dimensions (inches)
8.50000 w
11.00000 h

Print option: 
margin: none
paper size: letter
layout: portait

Printed page photo:
![avery 5160](https://cloud.githubusercontent.com/assets/3325882/18209662/be1df550-7102-11e6-81a5-9267c2023e76.jpg)

The first column seems good, but the second and third column spacing error(I tried 0.1 in label  horizontal spacing and 0.3 in with the same result). Can someone help me with this weird issue?

**Comment by pyro106** [comment]
Hello All, I'm actually having the same issue.  Virtually the same settings (I am also printing out a 1d barcode alongside the QR code), but the label width does not seem to match.  I measured it out, thinking it was just the horizontal spacing, but the actual width of the label is not the same as what I have put in the settings.  

NOTE: I am attempting to configure the settings to match the Avery 5160 Label sheet.

**Comment by Allenvh** [comment]
I am having the same issue. Have anyone found a workaround? 

Anyone printing labels on a dymo printer?

**Comment by Allenvh** [comment]
SUGGESTION!!!
wouldn't be better if the bulk labels are generated in a PDF that you can download?

**Comment by nrasmus** [comment]
Also having this issue--subscribing for updates.

**Comment by fanta8897** [comment]
Had the same issue, meddled with the settings till it worked. Also, I love that PDF label suggestion! We can then even move them down if we are not using a whole paper or something of that sort!

**Comment by nrasmus** [comment]
@fanta8897 can you share your settings?

**Comment by sau70networkadmin** [comment]
@fanta8897 I'd love to see those settings too, please!

**Comment by fanta8897** [comment]
@nrasmus @sau70networkadmin 
Whoops, sorry did not see your comments.
The settings I have are *almost* perfect. They are tight on the edges and sometimes the last barcode line can be cut off, but that does not matter as the last two lines are redundant terminating barcode lines. Nothing else is cut off and they are snug against the edges. Make 100% sure not to use a margin or headers and footers in the print options, they mess things up. Lastly, I have tested and setting the paper to black and white/media to labels, significantly reduces the amount of ink that rubs off.
![image](https://cloud.githubusercontent.com/assets/4807843/22572659/1b5cedbe-e973-11e6-96c8-5fd0c1787220.png)

![image](https://cloud.githubusercontent.com/assets/4807843/22572656/16635a64-e973-11e6-8ca9-cb146358a3f7.png)

**Comment by nrasmus** [comment]
Thank you @fanta8897  !  Not perfect, but close enough to use, which is all I really needed.

**Comment by sau70networkadmin** [comment]
Yes thank you @fanta8897 ! Also got me close enough. What I ended up doing was jiggering your numbers slightly, and printing only 15 labels per page -- then flip the sheet of labels 180° and print the second page, so that the "drift" doesn't matter. Details posted here: 
http://whoopis.com/core/snipe-it-printing-avery.html

**Comment by fbe-it** [comment]
I wanted to jump in here as I have also found a solution for the [Sheets of EconoGuard™ Vinyl Labels - ¾" x 2" (50 Labels / Sheet)](http://www.myassettag.com/Asset-Tags/Laser-Printable-Labels/SKU-LZR-AT-VINYL-A.aspx).

<img width="820" alt="snipe it - asset tag sheet label parameters - final" src="https://cloud.githubusercontent.com/assets/22353756/23107365/e90f62a6-f6b0-11e6-9e3a-32af9ef1cee8.png">

**Comment by averyfan** [comment]
Either Avery needs to make its template work with Epson ET-2750 printers or Epson needs to make its model ET-2750 printers work with Avery templates.

A Windows 10 computer using both Microsoft Word program and LibreOffice Writer program will correctly run an Avery 5160 template that displays a good Print Preview page, but fail to correctly print that page on an Epson ET-2750.  The failed printout result has all 30 labels with almost blank address line 1, a smudge partial print of address line 2, and a solid good print of address line 3.   A work around fix is to make a screen shot jpg of the top of the page and bottom of the page, then paste those on to a new blank page in Word or Writer and if you did a good job of lining up the jpg's, you can print a 5160 label page.

This was for printing 30 labels with the same address. I did not try the Word program feature of merge printing 30 different addresses from an Excel spreadsheet.  I did not try printing on other printers.

**Comment by caikems** [comment]
So I can only get these to line up for a single page print. I used the recommended settings and set it to 15 labels because anymore than 15 and everything starts to get unaligned. 
However, If I am trying to print multiple pages, the top margin is different on page one than every other page in the print job.

Here is the top margin on page 1 - The labels line up great and print properly:
![labels page1](https://user-images.githubusercontent.com/6711405/58721330-8d8b8500-8389-11e9-8b3e-c10cf1433370.PNG)

Here is the top margin on every page from page 2 until the last - The labels are all off alignment because of this huge top margin coming out of nowhere.
![labels page2](https://user-images.githubusercontent.com/6711405/58721331-8d8b8500-8389-11e9-8a16-0c494ef72c0a.PNG)

Any suggestions? Snipe-IT has been so awesome thus far. Just need to get the labels printing correctly and then we're set!

**Comment by ricoxg** [comment]
I ran into this same issue and I think I got it drilled just a little bit tighter for the Avery 8160 labels.

This allowed me to get 30 labels per page and while the third column was super tight and the bottom row not quite where I'd like it, I was able to print and all 30 were usable.

NOTE:  Key part of accomplishing this was to disable margins in the print menu and use just the margin adjustments in Snipe-IT

![snipe_labels](https://user-images.githubusercontent.com/42521869/66437442-58560a00-e9ef-11e9-8b87-13e35da4c824.png)

**Comment by brianrking** [comment]
We're also having this issue.  I'd love a less tetchy way to bulk-print labels.  It seems very much like you can get things sort-of, close to right, as long as you don't breath on it after you've configured it. This a big barrier to entry for us - we've spent longer trying to get the labels right than we did setting up the server, installing & configuring the software, and configuring & testing import and inventory scripts.  It will also be an ongoing issue, as we'll have to deal with it every summer when we get new equipment in and need to print a couple of hundred labels.

I wish I knew more about html & pdf printing to be able to understand exactly where the issue lays, but it certainly seems that something about the layout algorithm for labels is working in an unexpected way (at least unexpected to the layperson).

**Comment by HacDan** [comment]
We finally sat down and figured out settings that will work for printing 30 labels at a time and I wanted to comment for others to have these settings during their Googling!

**Labels per page:** _30_
**Label font size:** _8_
**Label dimensions:** _2.8125 w, 1.025 h_
**Label spacing:** _0.25 horizontal, 0.05 vertical_
**Page margins:** _0.5 top, 0.0 bottom, 0.0 right, 0.35 left_
**Page dimensions:** _8.5 w, 11 h_

I hope that helps someone down the road, probably myself!

**Comment by GamingSanctum** [comment]
> We finally sat down and figured out settings that will work for printing 30 labels at a time and I wanted to comment for others to have these settings during their Googling!
> 
> **Labels per page:** _30_ **Label font size:** _8_ **Label dimensions:** _2.8125 w, 1.025 h_ **Label spacing:** _0.25 horizontal, 0.05 vertical_ **Page margins:** _0.5 top, 0.0 bottom, 0.0 right, 0.35 left_ **Page dimensions:** _8.5 w, 11 h_
> 
> I hope that helps someone down the road, probably myself!


@HacDan 
Are you printing Avery 5160? What Browser are you printing from? Default margins on your printer options? 
I tried these settings and they were still WAY off for me.

**Comment by HacDan** [comment]
No margins on the printer or whatever the chrome minimal settings are. 


We found this to be very printer specific as we swapped our HP MFPs for Kyocera units and had to redo this. 

Takes a couple hours but saves time in the long run.

**Comment by dthompson0627** [comment]
> 

When I try to print with these settings the application shrinks the columns to 2 instead of 3. Do you have any idea why this is happening? 

![image](https://user-images.githubusercontent.com/9028594/199108094-ccdcbd48-d3ff-4782-b677-1f534011327c.png)

**Comment by dthompson0627** [comment]
> 
Do you have any idea why with these settings the application will shrink the columns available from 3 to 2? When I use these settings I can't actually get three to line up in a row.

**Comment by HacDan** [comment]
@dthompson0627 Without coming across too troll-like, I believe 3 (2.8125 + 0.25) > 8.5

So you have a quarter inch spacing on top of the 2.8125 inch wide labels. So, the math just doesn't add up to be less than 8.5 which is what your page width is currently set to. 

I'm guessing you were thinking it's only adding 0.5 inch as it's the spacing on either side of the center label, but it just doesn't work like that unfortunately. 

I hope that helps!

**Comment by dthompson0627** [comment]
You are not coming across as a troll at all! I'm really struggling with this so I appreciate all the help I can get.

I modified the numbers to make them 2.5 and .1 but that didn't work. When I run the calc it comes out as 7.8 which is less than 8.5 but it's still 2 per row.
![image](https://user-images.githubusercontent.com/9028594/199237001-0d188aa8-5902-44b9-97ea-90e5c27aa2c0.png)
![image](https://user-images.githubusercontent.com/9028594/199237465-a45999ba-0149-4a4f-b6ca-a7f27fa4fca9.png)


I don't understand what settings I'm getting wrong here, Even when I match what @ch-gz has I still get the same 2 per row. I feel like I'm missing something but haven't been able to determine what that is.

**Comment by HacDan** [comment]
@dthompson0627 All I'm seeing now is your left and right margins being quite large. 

My math works out to be ~8.3 inches or so, which may be too close to "edgeless" printing that most printers do not support. Not sure. 

Don't forget the label spacing is counted 3 times, so it works out to be (_label width_ * 3) + (_label spacing_ * 3) + _Left Margin_ + _Right Margin_

**Comment by palanthas** [comment]
Sadly, none of the above measurements worked and since my OCD wouldn't accept only printing 15 labels, I broke out my digital calipers to measure everything. I have my measurements nearly perfect on Windows 11, Chrome and an HP LaserJet Pro 4001dne printing on Avery 5160.

I note my OS, browser and printer model as I can't guarantee that these measurements will work for anyone else or on a different setup.

**Snipe IT setup**
![image](https://github.com/snipe/snipe-it/assets/5349049/277e8a42-20e9-4ff3-8dfd-e7db69dbf32c)

**Google Chrome setup**
![image](https://github.com/snipe/snipe-it/assets/5349049/6edb6daf-9f96-49a1-8b89-51653e70723b)

Avery 5160 labels are technically 1 inch tall but I found that I had issues with the bottom barcode drifting onto the next label so I opted for .99 inches and then a .01 label spacing. Between that and no bottom margin, everything fits well.

**NOTE: I have not tried multipage yet. Just a single 30 item page.**
